### PR TITLE
 Comment's visibility option values should be generated with the same format #2421 

### LINKED ISSUE #2421: Comment's visibility option values should be generated with the same format
now after edit a comment's visibility options,
its option values (generated by javascript) have no space in between.

but after refresh,
the values (generated by front-end java codes) will have spaces in between like "GIVER, STUDENTS, INSTRUCTORS".

This different behaviours will fail the test, see commit c7b7ae8

### PR DESCRIPTION
**Pull request by chowyb**
Fixes #2421

### CODE REVIEW

**Comment by chowyb** [comment]
Current fix is to standardize the values to be separated by `, ` (comma and whitespace), which was produced by the original Java code.

**Comment by damithc** [comment]
Add a comment when ready for review.

**Comment by chowyb** [comment]
Previous tests may need to be updated via GodMode despite it passing (ref commit [c7b7ae8](https://github.com/TEAMMATES/teammates/commit/c7b7ae88605bb6100db30a7bbdd173049a53890a)), but unsure if this should be done as part of this issue.

**Comment by chowyb** [comment]
Ready for review

**Comment by JosephineKwa** [comment]
You could update the tests here as well as they are relevant to this PR

**Comment by JosephineKwa** [comment]
The comment visibility options on some other pages are not in the same format, for example, in `instructorStudentRecordsPage`. Try to locate the other pages where this issue is present, and rectify them as well.

**Comment by JosephineKwa** [comment]
Is this ready for review?

**Comment by chowyb** [comment]
Was busy looking through the current tests, turns out they do not require updating via GodMode as they only test for one value at a time, and do not run into any problems with this.

Should I create additional tests for this issue?

**Comment by JosephineKwa** [comment]
It would be good to, since there are no tests checking the issue that was handled in this PR

**Comment by chowyb** [comment]
Update: found another location where the format is different (new response comment), will be working on that for now.

**Comment by chowyb** [comment]
On hold due to an issue with testing, will require #5383 to be resolved before proceeding.

File: src/main/java/teammates/ui/controller/PageData.java [hunk 1]
```diff
@@ -952,7 +952,12 @@ public boolean isResponseCommentGiverNameVisibleTo(FeedbackResponseCommentAttrib
     }
     
     public String getResponseCommentVisibilityString(FeedbackQuestionAttributes qn) {
-        return "GIVER," + StringHelper.removeEnclosingSquareBrackets(qn.showResponsesTo.toString());
+        String visibilityString = StringHelper.removeEnclosingSquareBrackets(qn.showResponsesTo.toString());
+        if (StringHelper.isWhiteSpace(visibilityString)) {
+            return "GIVER";
+        } else {
+            return "GIVER, " + visibilityString;
+        }
```
Comment (JosephineKwa) [inline src/main/java/teammates/ui/controller/PageData.java]: See if you could write this more concisely, perhaps, 
```
return StringHelper.isWhiteSpace(visibilityString) ? "GIVER" : "GIVER, " + visibilityString;
```

**Comment by JosephineKwa** [comment]
Looks good to merge, pending @damithc 's approval

**Comment by JosephineKwa** [comment]
Just a minor change requested

**Comment by chowyb** [comment]
@JosephineKwa ready for review! :)

**Pull request merged**